src/sagemaker/estimator.py

```diff
@@ -695,14 +695,19 @@ def _stage_user_code_in_s3(self) -> str:
 
         Returns: S3 URI
         """
-        local_mode = self.output_path.startswith("file://")
+        local_mode = not is_pipeline_variable(self.output_path) and self.output_path.startswith(
```

Comment:
I guess this works, but the statement is confusing. Whether it is local mode shouldn't depends on whether the output path is a pipeline variable or not. 

The follow up if elif block is really hard to read. 

How about just do this
```
if is_pipeline_variable(self.output_path):
    if self.code_location is None:
         ...
    else:
         ...
else: 
```